### PR TITLE
fix extended help for Doom 2, load help screen from PWAD

### PR DESCRIPTION
**Pull request by rfomin**
Inspired by this thread [[doomworld]](https://www.doomworld.com/forum/topic/111465-boom-extended-help-screens-an-undocumented-feature/) (thanks, Redneckerz), I've fixed the extended help screens for Doom 2. Tested with sherlock.wad

Also, I think we should load help screen from PWADs, they are useful sometimes (e.g. DBP27.wad [[idgames]](https://www.doomworld.com/idgames/levels/doom2/Ports/d-f/dbp27))

### CODE REVIEW

File: Source/m_menu.c [hunk 1]
```diff
@@ -4027,12 +4027,10 @@ void M_InitExtendedHelp(void)
 	{
 	  if (extended_help_count)
 	  {
-	    if (gamemode == commercial)
-	      {
-		ExtHelpDef.prevMenu  = &ReadDef1; // previous menu
-		ReadMenu1[0].routine = M_ExtHelp;
-	      }
-	    else
+	    ExtHelpDef.prevMenu  = &HelpDef; // previous menu
```
Comment (fabiangreffrath) [inline Source/m_menu.c:4030]: Why did you change the `if (gamemode == commercial) [...] else` logic here?

Comment (rfomin) [reply Source/m_menu.c:4030]: Because both Doom 2 and Doom 1 display a help screen if the user presses F1. For some reason, extended help was only available from the "Read This!" menu in Doom 1.

Comment (fabiangreffrath) [reply Source/m_menu.c:4030]: Please add a comment here.

File: Source/m_menu.c [hunk 2]
```diff
@@ -4379,10 +4377,23 @@ int M_GetPixelWidth(char* ch)
 
 void M_DrawHelp (void)
 {
+  int helplump;
```
Comment (fabiangreffrath) [inline Source/m_menu.c:4380]: Please add a comment here.

**Review by fabiangreffrath — CHANGES_REQUESTED** [review]

**Comment by fabiangreffrath** [comment]
Thank you!
